MIGRATING.md

```diff
@@ -0,0 +1,77 @@
+   - `getLogs(java.lang.String, java.lang.String, java.lang.String, java.lang.String, java.lang.String)`
+   Signature changed to `getLogs(java.util.Date, java.util.Date, java.lang.String, java.lang.String, java.lang.String)`
+   
+- `com.okta.sdk.resource.group.GroupList` 
+   - `listGroups(java.lang.String,java.lang.String, java.lang.String)`
+   Signature changed to `listGroups(java.lang.String, java.lang.String)`
+
+- `com.okta.sdk.resource.application.JsonWebKey`
+   - `getExpiresAt` Return type changed from `Date` to `String`
+
+- `com.okta.sdk.resource.application.SwaThreeFieldApplicationSettingsApplication`
+   - `getTargetUrl` Renamed to `getTargetURL`
+   - `setTargetUrl` Renamed to `setTargetURL`
+
+- `com.okta.sdk.resource.group.Group`
+   - `getType` Return type changed from `String` to `com.okta.sdk.resource.group.GroupType`
+
+- `com.okta.sdk.resource.group.rule.GroupRule`
+   - Removed `delete` 
+   - Removed `getAllGroupsValid` 
+   - Removed `getEmbedded` 
+   - Removed `setAllGroupsValid`
+
+- `com.okta.sdk.resource.log.LogAuthenticationContext`
+   - `getCredentialProvider` Return type changed from `List` to `com.okta.sdk.resource.log.LogCredentialProvider`
+   - `getCredentialType` Return type changed from `List` to `com.okta.sdk.resource.log.LogCredentialType`
+
+- `com.okta.sdk.resource.policy.Policy`
+   - Removed `createRule`
+
+- `com.okta.sdk.resource.policy.PolicyRule`
+   - Removed `setId`
+
+- `com.okta.sdk.resource.user.factor.VerifyFactorRequest`
+   - Removed `getTokenLifetimeSeconds` & `setTokenLifetimeSeconds`
+
+- `com.okta.sdk.resource.user.Role`
+   - `getType` Return type changed from `String` to `com.okta.sdk.resource.role.RoleType`
+   - `setType` Param type changed from `String` to `com.okta.sdk.resource.role.RoleType`
+
+- `com.okta.sdk.resource.user.User`
+   - Removed `addFactor`
+   - Renamed `addGroupTarget` to `addGroupTargetToRole`
+   - Renamed `addRole` to `assignRole`
+   - Removed `endAllSessions`
+   - `expirePassword` Return type changed from `com.okta.sdk.resource.user.TempPassword` to `com.okta.sdk.resource.user.User`
+   - Removed `forgotPassword`
+   - `getFactor` Return type changed from `com.okta.sdk.resource.user.factor.Factor` to `com.okta.sdk.resource.user.factor.UserFactor`
+   - Removed `listAppLinks`
+   - `listFactors` Return type changed from `com.okta.sdk.resource.user.factor.FactorList` to `com.okta.sdk.resource.user.factor.UserFactorList` 
+   - Renamed `listGroupTargetsForRole` to `listGroupTargets`
+   - Removed `listRoles`
+   - `listSupportedFactors` Return type changed from `com.okta.sdk.resource.user.factor.FactorList` to `com.okta.sdk.resource.user.factor.UserFactorList`
+   - Renamed `removeGroupTargetFromRole` to `removeGroupTarget`
+   - `resetPassword(java.lang.String, java.lang.Boolean)` Signature changed to `resetPassword(java.lang.Boolean)`
+
+- `com.okta.sdk.resource.user.UserCredentials`
+   - Removed `getEmails` & `setEmails`
```

Comment:
why?

Reply:
property `emails` is laid to rest in the new spec. Apparently, `emails` is no longer part of `UserCredentials`.